homeassistant/components/august/activity.py

```diff
@@ -107,12 +95,9 @@ def async_get_latest_device_activity(self, device_id, activity_types):
             if updated_device_ids:
                 for device_id in updated_device_ids:
                     _LOGGER.debug(
-                        "async_dispatcher_send (from activity stream): AUGUST_DEVICE_UPDATE-%s",
-                        device_id,
-                    )
-                    async_dispatcher_send(
-                        self._hass, f"{AUGUST_DEVICE_UPDATE}-{device_id}"
+                        "signal_device_id_update (from activity stream): %s", device_id,
                     )
+                    self.signal_device_id_update(device_id)
```

Comment:
You cannot call this here as it's not async safe.

Reply:
Also, why call `_process_newer_device_activities` if we get a RequestException? I would expect to see a `continue` inside the `except`.

Reply:
@balloob 
I'm having trouble determining what isn't safe about it.  Can you help point me in the right direction?
```
    def signal_device_id_update(self, device_id):
        """Call the callbacks for a device_id."""
        if not self._subscriptions.get(device_id):
            return

        for update_callback in self._subscriptions[device_id]:
            self._hass.loop.call_soon_threadsafe(update_callback)
```

Reply:
Fixed the `_process_newer_device_activities` not being skipped

Reply:
threadsafe usually means it's protected by a lock to ensure you're single source mutating data at a point in time. Inside async, locking on such a thing can cause a deadlock, as the task blocks the event loop until the resource is available, but it might be held by another task on the event loop that now can't continue.

Reply:
I think I've fixed this in https://github.com/home-assistant/home-assistant/pull/32249/commits/085648dee8335720844ea0445f60f4252b8256fc but it needs another look when you can

Reply:
Thanks.  Fixed